cli/azd/extensions/azure.foundry.ai.agents/internal/project/parser.go

```diff
@@ -0,0 +1,1291 @@
+		// Create role assignment
+		parameters := armauthorization.RoleAssignmentCreateParameters{
+			Properties: &armauthorization.RoleAssignmentProperties{
+				RoleDefinitionID: to.Ptr(fullRoleDefinitionID),
+				PrincipalID:      to.Ptr(principalID),
+			},
+		}
+
+		resp, err := client.Create(ctx, scope, roleAssignmentName, parameters, nil)
+		if err != nil {
+			// Check if the error is due to role assignment already existing (409 Conflict)
+			if strings.Contains(err.Error(), "RoleAssignmentExists") || strings.Contains(err.Error(), "409") {
+				fmt.Println("✓ Role assignment already exists (detected during creation)")
+				assignmentExists = true // Mark as existing so we skip waiting
+			} else {
+				return fmt.Errorf("failed to create role assignment: %w", err)
+			}
+		} else {
+			fmt.Println("✓ Successfully assigned 'Azure AI User' role")
+
+			if resp.Name != nil {
+				fmt.Printf("  Assignment ID: %s\n", *resp.Name)
+			}
+		}
+	}
+
+	// Only wait for propagation if we just created a new assignment
+	if !assignmentExists {
+		fmt.Println()
+		fmt.Println("⏳ Waiting 30 seconds for RBAC propagation...")
+		time.Sleep(30 * time.Second)
+
+		// Validate the assignment
+		fmt.Println("Validating role assignment...")
+		pager = client.NewListForScopePager(scope, &armauthorization.RoleAssignmentsClientListForScopeOptions{
+			Filter: to.Ptr("atScope()"),
+		})
+
+		validated := false
+		for pager.More() {
+			page, err := pager.NextPage(ctx)
+			if err != nil {
+				fmt.Fprintln(os.Stderr, "Warning: Could not validate role assignment. It may still be propagating.")
+				break
+			}
+
+			for _, assignment := range page.Value {
+				if assignment.Properties != nil && assignment.Properties.RoleDefinitionID != nil {
+					if strings.Contains(*assignment.Properties.RoleDefinitionID, roleDefinitionID) {
+						fmt.Println("✓ Role assignment validated successfully")
+						fmt.Printf("  Role: Azure AI User\n")
+						validated = true
+						break
+					}
+				}
+			}
+			if validated {
+				break
+			}
+		}
+
+		if !validated {
+			fmt.Fprintln(os.Stderr, "Warning: Could not validate role assignment. It may still be propagating.")
+		}
+	}
+
+	return nil
+}
+
+// deactivateHelloWorldRevision deactivates the hello-world placeholder revision using Azure SDK
+func deactivateHelloWorldRevision(ctx context.Context, resourceID string) error {
+	fmt.Println()
+	fmt.Println("======================================")
+	fmt.Println("Deactivating Hello-World Revision")
+	fmt.Println("======================================")
+	fmt.Println("ℹ️  Azure Container Apps requires an image during provision, but with remote Docker")
+	fmt.Println("   build, the app image doesn't exist yet. A hello-world placeholder image is used")
+	fmt.Println("   during 'azd provision', then replaced with your app image during 'azd deploy'.")
+	fmt.Println("   Now that your app is deployed, we'll deactivate the placeholder revision.")
+	fmt.Println()
```

Comment:
Still unclear why we need this... We don't do anything like this for regular ACA that follows the same flow.